### ISSUE TITLE
Code is getting injected into other files open in VSCode leavng them corrupt

### ISSUE THREAD

**Issue by stephen147** [Autodesk-AutoCAD/AutoLispExt#248]
**Describe the bug**
Code is getting injected into other files. When debugging and executing a file in VSCode, portions of the code are getting injected into other open files.

**To Reproduce**
1. Open a file and other files.
2. Go to the file you want to debug and run in ACAD.
3. Run the code.
4. This is where I don't know, why or how it happens but the code that I'm debugging gets injected into the other open files and they are saved leaving them corrupt.

**Expected behavior**
For step 4 above not to happen.

**Desktop (please complete the following information):**
 - OS: Windows 10 Pro (Build 19045)

**Comment by EnM4st3r** [comment]
im always working and debugging with multiple files open, never experienced something similar

**Comment by stephen147** [comment]
I found the issue. When you use the find and replace all in VSCode, this happens. It doesn't open the file and replaces more than it should.

Perhaps, this is associated with this extension so I might leave this open for others to test.

**Comment by stephen147** [comment]
Nope. I didn't do any replacements this time while I had VSCode open, so this bug is still there.

It's when I have a file open and execute it and then tab over to another tab to modify a different lisp file or something like that. 

I don't know when this exactly happens. I'm trying to find an exact way to reproduce this bug.

**Comment by Djwhitelion** [comment]
Check your defun names. If there is a similar defun name and that file is loaded in autocad. It can load the other defun in a loaded lisp instead. If you run similar code in other files. And call them the same. They can be called between them.

**Comment by stephen147** [comment]
These are the steps to reproduce the bug at least for me anyway.

1. Open file A with the contents:
```
(defun test () 
  (princ "Test")
)

(test)
```
2. Open File B with the contents:
```
(defun test () 
  (princ " Me")
)

(test)
```
3. Run the code on file B.
4. Change the defun name test to `testme` on both File A and File B. In my experience change any part of the code will have the same effect.
5. Switch back to file A and evaluate the code.
6. File B now gets injected with the code.

For step 5, the same happens when you save the file. This only happens sometimes.

**Comment by chris-harling** [comment]
This is certainly an issue for me. From what I can tell it seems to do with committing changes. It does not seem to be a find & replace or similar named function issue for me. I'm getting entire 600+ line files getting overwritten with other files.
I've only noticed it to be an issue with whatever files were open at the time. Typically for me this is a few .lsp scripts and the working tree.

I'll try to narrow it down and see if I can make it repeatable instead of hearsay.

One sort of workaround I've found is ctrl+z will undo the changes in a single step.

**Comment by jennbushey** [comment]
When I have two .lsp files open side by side in VS Code, editing one file causes it to be overwritten by the other file when the focus changes to the other file.

For example:
_somefile.lsp_ has 50 lines of code and _anotherfile.lsp_ has 75. Editing _anotherfile.lsp_ and then switching focus to _somefile.lsp_ causes the first 50 lines of _anotherfile.lsp_ to be replaced by the contents of _somefile.lsp_. The whole code is replaced if _somefile.lsp_ is the longer code.

Workaround: Change the language mode of one or both files to "Plain Text"

**Comment by GoCoder7** [comment]
I can reproduce the same family of corruption behavior, and I found what looks like a very strong root cause in the extension code.

I originally suspected another AutoLISP extension in my setup, but after tracing the edit paths, the bug appears to be in AutoLispExt's formatter implementation itself.

In the installed extension bundle, the document formatter is implemented roughly like this:

```js
provideDocumentFormattingEdits(document) {
  let activeTextEditor = vscode.window.activeTextEditor;
  if (activeTextEditor == undefined) return [];

  let fmt = LispFormatter.format(activeTextEditor.document, null);
  return [vscode.TextEdit.replace(utils.getFullDocRange(activeTextEditor), fmt)];
}
```

The important problem is that it ignores the `document` parameter and instead formats `window.activeTextEditor.document`.

That means if formatting is invoked for file B while file A is the active AutoLISP editor, file B can be replaced with file A's content. That matches the reports in this issue very closely:
- one open `.lsp` file overwriting another
- corruption after focus changes / save / debug / run workflows
- `Ctrl/Cmd+Z` restoring the file in a single step

I also verified that another AutoLISP extension in my setup does not have a full-document background replace path, so this looks very specifically tied to AutoLispExt's formatter behavior.

In my setup, `[autolisp].editor.defaultFormatter` was set to `Autodesk.autolispext`, which makes this even more suspicious.

Possible trigger paths:
- Format Document / Format Selection
- format-on-save if enabled anywhere
- any debug/run/save workflow that ends up invoking the formatter for a non-active document

Workarounds that seem reasonable until this is fixed:
- disable AutoLispExt as the default formatter for AutoLISP
- disable format-on-save for AutoLISP
- Save All before debug/run
- temporarily switch affected files to Plain Text if needed

The formatter should use the `document` argument passed into `provideDocumentFormattingEdits` / `provideDocumentRangeFormattingEdits`, not `window.activeTextEditor.document`.
